File: openapi-generator/kotlin-spring/src/main/kotlin/org/openapitools/model/AssetReeferResponseReeferStats.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import org.openapitools.model.AssetReeferResponseReeferStatsAlarms1
import org.openapitools.model.AssetReeferResponseReeferStatsEngineHours
import org.openapitools.model.AssetReeferResponseReeferStatsFuelPercentage
import org.openapitools.model.AssetReeferResponseReeferStatsPowerStatus
import org.openapitools.model.AssetReeferResponseReeferStatsReturnAirTemp
import org.openapitools.model.AssetReeferResponseReeferStatsSetPoint
import javax.validation.Valid
import javax.validation.constraints.*

/**
 * 
 * @param fuelPercentage Fuel percentage of the reefer
 * @param powerStatus Power status of the reefer
 * @param engineHours Engine hours of the reefer
 * @param setPoint Set point temperature of the reefer
 * @param returnAirTemp Return air temperature of the reefer
 * @param alarms Reefer alarms
 */
data class AssetReeferResponseReeferStats (

        @JsonProperty("fuelPercentage") val fuelPercentage: List<AssetReeferResponseReeferStatsFuelPercentage>? = null,

        @JsonProperty("powerStatus") val powerStatus: List<AssetReeferResponseReeferStatsPowerStatus>? = null,

        @JsonProperty("engineHours") val engineHours: List<AssetReeferResponseReeferStatsEngineHours>? = null,

        @JsonProperty("setPoint") val setPoint: List<AssetReeferResponseReeferStatsSetPoint>? = null,

        @JsonProperty("returnAirTemp") val returnAirTemp: List<AssetReeferResponseReeferStatsReturnAirTemp>? = null,

        @JsonProperty("alarms") val alarms: List<AssetReeferResponseReeferStatsAlarms1>? = null
) {

}

File: openapi-generator/scala-gatling/src/gatling/scala/org/openapitools/client/model/AssetReeferResponseReeferStats.scala

package org.openapitools.client.model


case class AssetReeferResponseReeferStats (
    /* Fuel percentage of the reefer */
    _fuelPercentage: Option[List[AssetReeferResponseReeferStatsFuelPercentage]],
    /* Power status of the reefer */
    _powerStatus: Option[List[AssetReeferResponseReeferStatsPowerStatus]],
    /* Engine hours of the reefer */
    _engineHours: Option[List[AssetReeferResponseReeferStatsEngineHours]],
    /* Set point temperature of the reefer */
    _setPoint: Option[List[AssetReeferResponseReeferStatsSetPoint]],
    /* Return air temperature of the reefer */
    _returnAirTemp: Option[List[AssetReeferResponseReeferStatsReturnAirTemp]],
    /* Reefer alarms */
    _alarms: Option[List[AssetReeferResponseReeferStatsAlarms1]]
)
object AssetReeferResponseReeferStats {
    def toStringBody(var_fuelPercentage: Object, var_powerStatus: Object, var_engineHours: Object, var_setPoint: Object, var_returnAirTemp: Object, var_alarms: Object) =
        s"""
        | {
        | "fuelPercentage":$var_fuelPercentage,"powerStatus":$var_powerStatus,"engineHours":$var_engineHours,"setPoint":$var_setPoint,"returnAirTemp":$var_returnAirTemp,"alarms":$var_alarms
        | }
        """.stripMargin
}
